sys/hacl/build.rs

```diff
@@ -30,6 +38,7 @@ fn includes(home_dir: &Path, include_str: &str) -> Vec<String> {
     ]
 }
 
+// FIXME: drop all the host specific checks and move them to target.
 #[cfg(all(
     any(target_arch = "x86", target_arch = "x86_64"),
     any(
```

Comment:
Isn't `target_arch` meant to check for target arch?

Reply:
All these checks are compile time checks right now. But compile time of this build script, not the library. Hence this is wrong, because it's not for the target, but the host.